contract/tests/test_contract.py

```diff
@@ -483,22 +504,35 @@ def test_onchange_partner_id(self):
             self.contract.partner_id.property_product_pricelist,
         )
 
+    def test_onchange_partner_id_domain(self):
+        with Form(self.contract) as contract_form:
+            contract_form.invoice_partner_id = self.partner_2
+        self.assertNotEqual(
```

Comment:
@pedrobaeza Both changes done and functional in views.

The first test for uom is ok, but this one does not seem to work. Do you have an idea?

Reply:
Are you trying to test domains? I think that through direct assignation, there's no domain restriction applied.

Reply:
In uom the behaviour is working in test. If you try to assign a wrong uom, it bypasses it.

Reply:
@pedrobaeza Or do you know the property of a field that has the evaluated domain value ?

Reply:
I have never tested such domains being frank. The only way I know right now is to call fields_view_get and check the domain attribute in the resulting xml. I can investigate later.

Reply:
Ok, I have found the way to test it. Set the invoice_partner_id to void before

Reply:
So, this one is ready

Reply:
Great, TIL about this way of testing domains through Form